src/Sylius/Bundle/CoreBundle/Doctrine/ORM/ChannelPricingRepository.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\Doctrine\ORM;
+
+use Sylius\Bundle\ResourceBundle\Doctrine\ORM\EntityRepository;
+use Sylius\Component\Core\Repository\ChannelPricingRepositoryInterface;
+
+class ChannelPricingRepository extends EntityRepository implements ChannelPricingRepositoryInterface
+{
+    public function findWithDiscountedPrice(): array
+    {
+        return $this->createQueryBuilder('o')
+            ->andWhere('o.originalPrice IS NOT NULL')
```

Comment:
Because of that, I've added an additional check if the applied promotions are not empty: https://github.com/Sylius/Sylius/pull/13100/files#diff-5f3a950780ee137f6c0dbe75d71b3ef264f25fbb5d634b9ec463fdab7b1b5819R37 
Wouldn't it be enough for now?

Reply:
Fair enough for the merge, but would be good to add that `where` too in the nearest future 🖖